AzureDevOpsPipelines/deployment/bot-publish/source/core.js

```diff
@@ -0,0 +1,96 @@
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
+const tl = require("azure-pipelines-task-lib");
+const child_process_1 = require("child_process");
+const path = require("path");
+const fs = require("fs");
+class core {
+    constructor() {
+        this.isLoggedIn = false;
+        this.cliPasswordPath = "";
+        this.servicePrincipalId = "";
+        this.servicePrincipalKey = "";
+        this.prefix = "";
+        this.subscriptionID = "";
+        this.getPrefix();
+        this.subscriptionID = tl.getVariable('subscriptionID');
+        if (this.subscriptionID == null) {
+            this.loginAzureRM(tl.getInput("connectedServiceNameARM", true));
+            tl.setVariable('subscriptionID', this.subscriptionID);
+        }
+    }
+    loginAzureRM(connectedService) {
+        var authScheme = tl.getEndpointAuthorizationScheme(connectedService, true);
+        this.subscriptionID = tl.getEndpointDataParameter(connectedService, "SubscriptionID", true);
+        if (authScheme.toLowerCase() == "serviceprincipal") {
+            let authType = tl.getEndpointAuthorizationParameter(connectedService, 'authenticationType', true);
+            let cliPassword;
+            var servicePrincipalId = tl.getEndpointAuthorizationParameter(connectedService, "serviceprincipalid", false);
```

Comment:
change type to const and add space below this line. Also, group all variables within the same type.